File: BookshelfAPIs/BookshelfAPIs/Models/book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace BookshelfAPIs.Models
{
    public class Book
    {
        public string Image { get; set; }
        public string Title { get; set; }
        public string Author { get; set; }
        public string Format { get; set; }
        public double Rating { get; set; }
        public double Price { get; set; }
        public double OldPrice { get; set; }
        public string ISBN { get; set; }
        public string Category { get; set; }


    }
}

File: bookshelf-frontend/src/app/models/book.ts
export class Book {
    Title?:string
    Author?:string
    Format?:string
    ISBN?:string
    Rating?:string
    Price?:number
    OldPrice?:number
    Image?:string
    Category?:string

    makeBook(template:Book){
        this.Title = template.Title
        this.Author = template.Author
        this.Format = template.Format
        this.ISBN = template.ISBN
        this.Rating = template.Rating
        this.Price = template.Price
        this.OldPrice = template.OldPrice
        this.Image = template.Image
        this.Category = template.Category
        return this
    }
}
